### PR TITLE
Changing fonts

### PR DESCRIPTION
**Pull request by DAlexIV**
Feature was implemented according to https://github.com/connectbot/connectbot/issues/458 request. For now, I've added only Hack font, but expanding font base shouldn't be a big deal, just put .ttf file in assets/fonts folder and add its name in FontManager. 
Also there could be [Downloadable fonts](https://developer.android.com/guide/topics/ui/look-and-feel/downloadable-fonts.html), but we need to upgrade minsdk to 14 to use them in support library.
And last but not least, I've added a font picker in host editing activity, so we need to translate two strings to all supported languages.
Looking for your feedback on this.

### CODE REVIEW

File: app/src/main/java/org/connectbot/util/FontManager.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.Locale;
+import java.util.Set;
+
+import android.content.Context;
+import android.graphics.Typeface;
+
+public class FontManager {
+	private static String FONT_DIRECTORY = "fonts";
+	public static String SYSTEM_FONT = "System";
+	private static Set<String> FONTS = new HashSet<>(Arrays.asList( "Hack"));
```
Comment (kruton) [inline app/src/main/java/org/connectbot/util/FontManager.java]: nit: space before `"Hack"`.

**Comment by fat-tire** [comment]
Just wanted to update this issue-- I'm working on something fun-- now that we're targeting sdk 27-- we can finally do [downloadable fonts](https://developer.android.com/guide/topics/ui/look-and-feel/downloadable-fonts) :)

Screenshots are on Chrome OS (Samsung Chromebook Pro).





(won't post all the mono fonts.  You get the idea.  And don't ask me why I started typing the numbers with "Z" in nano.  That's a typo, not an issue with the fonts.)

Some fonts such as VT323 have kerning turned on.  Is there an easy way to tell a Paint object not to do this?

This is pretty much done, though I haven't really done much testing to make sure everything works as expected.  I did update a bunch of libraries and such at the same time-- so I want to separate those changes out from the minimum required to make this work and verify that it's backwards compatible.  I think it should fallback for the -oss build as well-- the menu item simply won't appear and the default will be monospace.  Can easily incorporated the changes in this PR though if you want to bundle a few, but the downloadable fonts option adds virtually no size to the app (all fonts are cached by google play services).

Cheers.
ft

PS-- @DAlexIV I tried to use a few bits of your PR, but I didn't need stuff like the FontManager since that is handled by the FontRequest, and I had to change some stuff to make it asynchronous anyway to handle situations where for some reason the font isn't available (in which case it falls back to monospace).  I also didn't style the font options because it would require pre-downloading every one of them first.  But if you wanted to add bundled fonts, it would be very easy to do.

**Comment by tigerinus** [comment]
Any update?

**Comment by fat-tire** [comment]
See #613

**Comment by ebardie** [comment]
I use nerd fonts in my shell prompts, tmux furniture, etc., but ConnectBot's single font doesn't place nicely.

I know that switching to `termux`  is an option, but I'd prefer to stick with ConnectBot if only it could use custom fonts.
  
Given that this ticket was closed after a CLA issue, and that maybe the following would be worth considering to help unblock contributions?

https://ossbase.org/initiatives/cla-free/

**Pull request closed without merging**